### ISSUE TITLE
GenericLinuxProfile missing DPad buttons

### ISSUE THREAD

**Issue by ChrisNZL** [pbhogan/InControl#198]
Hey @pbhogan, bought InControl recently, really love it, thanks for making it!

One snag I've found - the GenericLinuxProfile does not contain any DPad button sources.
My Xbox 360 controller on Ubuntu 14 is reported as "Generic Controller"; everything works except for the DPad, which does not respond at all.

I created a gist with DPad button entries copied from the Xbox360LinuxProfile, as I know buttons 11-14 belong to the DPad (confirmed by editing entries on the Unity launcher's Input tab).

Would you please be able to merge this with InControl when you get the chance, if it looks okay?
https://gist.github.com/ChrisNZL/8ed3c4c54afbba390fbe

I'll try to see about making it a custom profile for now (though I'm not sure if the same naming will let it be picked up or not?)

Thanks

**Comment by ChrisNZL** [comment]
Upon further investigation, controller is actually being detected as "Xbox 360 Wireless Receiver (XBOX)" from `Input.GetJoystickNames`. Perhaps adding that to the `JoystickNames` array in Xbox360LinuxProfile.cs would suffice?

![image](https://cloud.githubusercontent.com/assets/280814/7103438/bf2d666c-e0fb-11e4-9dbc-7d74fedc89c5.png)

**Comment by pbhogan** [comment]
Hmm. I'm going to assume you're using Unity 5, because as of Unity 5 they're supposed to be using a new input system on Linux that maps all controllers to the same mappings (I was told this by Levi Bard, a Unity developer). So you'll notice the GenericLinuxProfile has a property `MinUnityVersion` set to Unity 5, and every other Linux profile is has `MaxUnityVersion` set to Unity 4.9. GenericLinuxProfile also uses `JoystickRegex` to catch all device names instead of specific names, again because all devices are supposed to have the same mappings now. 

I was told all this many moons ago, however, and perhaps they have made some adjustments to their mappings or haven't implemented it yet. I don't use Linux myself so I can't confirm any of this.

I'll add your DPad button entries to the generic profile, since you say it works. Could you fire up the TestInputManager scene and confirm all other controls work properly too?

**Comment by ChrisNZL** [comment]
Sorry, yes, using Unity 5.0.1. Works on both Ubuntu 14.10 and SteamOS.

I've just built the TestInputManager twice with and without the change and tested with Ubuntu for you, one last time to be sure:

- Without my change on the TestInputManager, the DPad does nothing, though everything else works.

- With my change, DPad Up/Down/Left/Right works, as does DPad X and Y, and can confirm that all other buttons and sticks and triggers work too, they all turn green.

I did try commenting out the `MaxUnityVersion` line in Xbox360LinuxProfile and adding my controller name to the array, but still it never got picked up.

So I guess with the "." regex check, picks up the controller fine as intended, GenericLinuxProfile was just missing the DPad button sources (since my DPad isn't being seen as analog for whatever reason).
